src/components/Pagination/PaginationRow.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import PaginationRowAmountOfPages from './PaginationRowAmountOfPages';
+import PaginationRowButtonGroup from './PaginationRowButtonGroup';
+import PaginationRowItems from './PaginationRowItems';
+import PaginationRowBack from './PaginationRowBack';
+import PaginationRowForward from './PaginationRowForward';
+import { PAGINATION_VIEW_TYPES, PAGINATION_VIEW } from './constants';
+import { Form, FormControl, FormGroup, ControlLabel } from '../Form';
+import { DropdownButton } from '../Button';
+import { MenuItem } from '../MenuItem';
+
+/**
+ * PaginationRow component for Patternfly React
+ */
+const PaginationRow = ({
+  className,
+  contentView,
+  viewType,
+  pagination,
+  amountOfPages,
+  itemCount,
+  itemsStart,
+  itemsEnd,
+  messages,
+  dropdownButtonId,
+  onPerPageSelect,
+  onFirstPage,
+  onPreviousPage,
+  onPageInput,
+  onNextPage,
+  onLastPage
+}) => {
+  const { page, perPage, perPageOptions } = pagination;
+  const classes = cx(
+    {
+      'content-view-pf-pagination': contentView,
+      'list-view-pf-pagination': viewType === PAGINATION_VIEW.LIST,
+      'card-view-pf-pagination': viewType === PAGINATION_VIEW.CARD,
+      'table-view-pf-pagination': viewType === PAGINATION_VIEW.TABLE,
+      clearfix: true
+    },
+    className
+  );
+  return (
+    <Form className={classes}>
+      <FormGroup>
+        <DropdownButton
+          title={perPage}
+          dropup
+          componentClass={PaginationRowButtonGroup}
+          onSelect={onPerPageSelect}
+          id={dropdownButtonId}
+        >
+          {perPageOptions &&
+            perPageOptions.length &&
+            perPageOptions.map((option, i) => {
+              return (
+                <MenuItem eventKey={option} active={option === perPage} key={i}>
+                  {option}
+                </MenuItem>
+              );
+            })}
+        </DropdownButton>
+        <span> {messages.perPage} </span>
+      </FormGroup>
+      <FormGroup>
+        <PaginationRowItems
+          itemCount={itemCount}
+          itemsStart={itemsStart}
+          itemsEnd={itemsEnd}
+          messagesOf={messages.of}
+        />
+
+        <PaginationRowBack
+          page={page}
+          messagesFirstPage={messages.firstPage}
+          messagesPreviousPage={messages.previousPage}
+          onFirstPage={onFirstPage}
+          onPreviousPage={onPreviousPage}
+        />
+
+        <ControlLabel className="sr-only">Current Page</ControlLabel>
+        <FormControl
+          className="pagination-pf-page"
+          type="text"
+          value={pagination.page}
+          onChange={onPageInput}
```

Comment:
Hitting enter on the input sends the form. I think we should avoid that.

Reply:
👍  good point - I can incorporate this. Since your component is not stateless, my tentative plan is to have it wrap my `PaginationRow`. I am thinking I will just call it `Paginator` since your's has some additional state and some API magic it is handling. I still think there is a lot of value in exposing the innermost pagination row components as stateless functional components. I will update my PR today to incorporate this...

Reply:
Yeah, I couldn't make it stateless with capturing this input's value.

If you expose all necessary callbacks then I think wrapping it with another component with the logic from my PR makes sense. It would be nice solution where we have all the benefits without loosing any flexibility, so :+1: 
We just need to make sure all required entry points are present. At the moment I think you'd need to add at least an `onFormSubmit` callback and it might be necessary to add separate prop for the input value.

Reply:
exactly my thought process... will update this PR shortly...